1. Basics/task_7/part_4/task_4_on_apache_handler.php

```diff
@@ -0,0 +1,188 @@
+<?php
+/**
+ * File to search logins and passwords
+ */
+define(constant_name: 'FILE_NAME', value: 'passwords.txt');
+
+processHttpRequest();
+
+// REQUEST PROCESS FUNCTIONS ///////////////////////////////////////////////////////////////////////
+/**
+ * @return void
+ */
+function processHttpRequest() : void {
+	try {
+        checkRequestMethod();
+        checkRequestUri();
+        checkRequestContentTypeValue();
+
+		$loginAndPasswordPair = validateLoginAndPassword();
+        checkFileExistance(FILE_NAME);
+
+		validateUser($loginAndPasswordPair['login'], $loginAndPasswordPair['password']);
+		echo '<h1 style=\'color:green\'>FOUND</h1>';
+    } catch(Exception $ex) {
+		http_response_code($ex->getCode());
+        echo '<p>' . $ex->getMessage() . '</p>';
+    };	
+}
+
+// REQUEST PROCESS FUNCTIONS. VALIDATON FUNCTIONS /////////////////////////////////////////////////
+/**
+ * @throws Exception
+ * @return void
+ */
+function checkRequestMethod() : void {
+	if(($_SERVER['REQUEST_METHOD'] ?? null) === 'POST') {
+		return;
+	}
+
+    throw new Exception('Request method is invalid. The method POST is needed.', 400);
+}
+
+/**
+ * @throws Exception
+ * @return void
+ */
+function checkRequestUri() : void {		
+	if(preg_match('#^/api/checkLoginAndPassword$#', $_SERVER['REQUEST_URI'] ?? '')) {
+		return;
+	}
+
+    throw new Exception('URI is invalid.', 404);
+}
+
+/**
+ * @throws Exception
+ * @return void
+ */
+function checkRequestContentTypeValue() : void {
+   	if(preg_match('#\bapplication/x-www-form-urlencoded\b#i', apache_request_headers()['Content-Type'] ?? '')) {
+		return;
+	}
+    
+    throw new Exception('Header\'s \'Content-Type\' value is invalid.', 400);
+}
+
+/**
+ * @throws Exception
+ * @return array
+ */
+function validateLoginAndPassword(): array {
+	$login = sanitizeInput($_POST['login'] ?? false);
+
+	if($login === false) {
+		throw new Exception('Login field is absent', 404);
+	}
+
+	if(empty($login)) {
+		throw new Exception('Login value is absent(empty field)', 404);
```

Comment:
and what is the difference between `if($login === false) {` and  `if(empty($login)) {`?